marlowe/test/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -419,6 +419,64 @@ choiceNotInBounds bounds =
     suchThat arbitrary $ \chosenNum -> not $ any (inBound chosenNum) bounds
 
 
+-- | Generate relevant Input content for a given input action
+interestingInput :: Bool -> Action -> [InputContent]
+interestingInput validity (Choice choiceId) bounds =
+  map (IChoice choiceId) (interestingChoiceNum bounds)
+  ++ map (\choiceId -> IChoice choiceId bounds) (interestingChoiceId choiceId)
+interestingInput validity (Choice choiceId) bounds = IChoice choiceId <$> interestingChoiceNum' validity bounds
+
+interestingChoiceId :: ChoiceId -> [ChoiceId]
+interestingChoiceId _ = error "not implemented yet"
+
+interestingParties :: Party -> [Party]
+interestingParties (Role "Alice") = [Role "Alice", Role "Alic", Role "Alices"]
+
+interestingChoiceNum :: [Bound] -> [ChoiceNum]
+interestingChoiceNum bounds = concatMap interestingChoiceNum' bounds
+
+interestingChoiceNum' :: Bool -> Bound -> [ChoiceNum]
+interestingChoiceNum' True (Bound lower upper)
+  | lower > upper = Nothing
+  | lower == upper =
+      [
+        lower
+      ]
+  | otherwise =
+      Data.List.nub
+      [
+        lower
+      , lower + 1
+      , upper - 1
+      , upper
+      , (lower + upper) `div` 2
+      ]
+
+interestingChoiceNum' False (Bound lower upper)
+  | lower > upper = Nothing
+  | lower == upper =
```

Comment:
I think I can collapse this case with the `otherwise` case below - looks like the same output